### PR TITLE
boards: 96b_carbon: Define bt irq and reset pins in dts

### PR DESCRIPTION
**Pull request by erwango**
Define reset-gpios and irq-gpios in bt-hci node.
This allows to remove definition of matching Kconfig symbols
from _defconfig file and remove unwanted warning.
Also remove CONFIG_BT_SPI which is also present in board
Kconfig.defconfig file.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

### CODE REVIEW

**Review by ydamigos — CHANGES_REQUESTED** [review]
@erwango You beat me to it. You should add the reset and irq properties to zephyr,bt-hci-spi yaml. I already have a patch ready if you want to use it  194cb55696e2bb3c9376cd0f6e40a6f25b295979

**Comment by ydamigos** [comment]
You could also define a bt alias to allow generation of defines with friendly names.

File: boards/arm/96b_carbon/96b_carbon_defconfig [hunk 1]
```diff
@@ -41,10 +41,3 @@ CONFIG_CLOCK_STM32_PLL_Q_DIVISOR=7
 CONFIG_CLOCK_STM32_AHB_PRESCALER=1
 CONFIG_CLOCK_STM32_APB1_PRESCALER=2
 CONFIG_CLOCK_STM32_APB2_PRESCALER=1
-
-# BT SPI configuration when BT is selected
-CONFIG_BT_SPI=y
```
Comment (ydamigos) [inline boards/arm/96b_carbon/96b_carbon_defconfig]: We should leave this.

Comment (erwango) [reply boards/arm/96b_carbon/96b_carbon_defconfig]: Why should we leave it? BT_SPI will be enabled in board Kconfig.defconfig it BT is enabled

**Comment by erwango** [comment]
> You beat me to it. You should add the reset and irq properties to zephyr,bt-hci-spi yaml. I already have a patch ready if you want to use it 194cb55

I've been fast but sloppy. Will do thanks.

File: boards/arm/96b_carbon/96b_carbon_defconfig (hunk 1 above)
Comment (ydamigos) [reply boards/arm/96b_carbon/96b_carbon_defconfig]: BT_SPI is not enabled in board Kconfig.defconfig right now.

**Comment by erwango** [comment]
Closing this PR as it is really bad. @ydamigos , please push yours

**Pull request closed without merging**